## Abstract {.page_break_before}

Genetic alterations in cancer perturb cellular function through diverse mechanisms.
Connecting these alterations to their downstream functional consequences is a vital step in precision medicine and biomarker identification efforts.
To link mutation status to dysregulation of specific cell processes, one can train a classifier to predict mutation presence or absence from features derived from -omics assays.

In this work, we compare data types profiled in the TCGA Pan-Cancer Atlas for use as a multivariate functional readout of point mutations and copy number variants (CNVs).
We focus on RNA sequencing, DNA methylation, reverse phase protein arrays (RPPA), microRNA, and somatic mutational signatures.
Since the genes that drive cancer development have varied functions, we anticipated that the most effective -omics assay would depend on the target gene.

RNA sequencing and DNA methylation were the most effective predictors of alteration state.
Surprisingly, we found that across a variety of cancer-associated genetic alterations, they were approximately equally effective predictors.
The target gene was the primary driver of performance, rather than the data type, and there was little difference between the top data types for the majority of genes.
We also found that combining data types into a single multi-omics model often provided little or no improvement in predictive ability over the best individual data type.
Based on our results, for the design of studies focused on the functional outcomes of cancer mutations, we recommend focusing on gene expression or DNA methylation as first-line readouts.


