## Methods

### Data download and preprocessing

1. Mutation calls (MC3) and copy number calls (GISTIC 2.0) from GDC; similar to Ras activation paper
2. Cancer gene information from Vogelstein 2013 Science driver gene paper, this includes oncogene/TSG predictions (they use activating vs. inactivating mutation frequency across cancer samples to predict this)
3. Omics data (RNA-seq, 27K/450K methylation, RPPA) from GDC/PanCanAtlas
4. Mutational signatures information for TCGA WES samples downloaded from ICGC data portal, originally generated in Alexandrov 2020 Nature paper
5. Sample filtering details the same as Ras activation paper
6. Gene expression was standardized per gene and filtered to top 8000 genes by MAD, methylation used raw beta values (sometimes compressed using PCA, see results section), RPPA was standardized per probe, mutational signatures used raw counts
8. Describe methylation data filtering/imputation, maybe also mention probe type correction experiments and point to supplementary figures?
7. How many TCGA samples were used in each experiment? (Can include Venn diagrams in supplement)

### Training classifiers to detect cancer mutations

1. Briefly describe elastic net logistic regression, point to Greg's papers
2. Describe stratified CV scheme (train/test sets have equal cancer type and sample type proportions)
3. Describe evaluation metrics (largely used AUPR, in most experiments used difference from shuffled baseline to correct for label imbalance)
4. Describe statistical testing (used t-tests to statistically compare distributions across CV splits, corrected for multiple testing using Benjamini-Hochberg as implemented in scipy, set a threshold of alpha = 0.001 to avoid false positives - FP rate can be estimated by looking at shuffled > signal examples)
5. Describe multi-omics experiments (just concatenated data matrices along sample axis, used intersection of samples between data types)
6. Describe tumor purity prediction experiments using elastic net linear regression (results in supplement)

### Data and software availability

1. Link to GitHub repo: https://github.com/greenelab/mpmp
2. This repository contains scripts to download and preprocess all the necessary data programmatically, in the `00_download_data`directory.
