## Results

### Using diverse data modalities to predict cancer alterations

We collected four different readouts from cancer samples in the TCGA Pan-Cancer Atlas, capturing four steps of cellular function that are perturbed by genetic alterations in cancer (Figure {@fig:overview}A).
These included gene expression (RNA-seq data), DNA methylation (27K and 450K Illumina BeadChip arrays), protein abundance (RPPA data), and patterns of somatic mutation (mutational signatures).
To link these diverse data modalities to changes in mutation status, we used elastic net logistic regression to predict the presence or absence of mutations in cancer genes, using these readouts as predictive features (Figure {@fig:overview}B).
We evaluated the resulting mutation status classifiers in a pan-cancer setting, preserving the proportions of each of the 33 cancer types in TCGA for all of 8 train/test splits (4 folds x 2 replicates) in each of 107 cancer genes (Figure {@fig:overview}C).

In our analyses, we sought to compare classifiers against a naive baseline (to identify genes whose mutation status correlates strongly with a gene signature in a given data type) and against one another (to identify data types that are more or less predictive of mutations in a given gene).
To account for variation between dataset splits in making these comparisons, we treat classification metrics from the 8 train/test splits as performance distributions, which we compare using _t_-tests.
For each comparison, we summarized performance across all genes in our cancer gene set using a volcano-like plot, in which each point is a gene.
The x-axis shows the magnitude of the change in the classification metric between conditions, and the y-axis shows the _p_-value for the associated _t_-test.
Figure {@fig:overview}C shows this evaluation pipeline.

![
blah blah figure legend
](images/figure_1.png){#fig:overview}

