## Results

### Overview (Figure 1)

1. Predicting mutation status from functional -omics data (something similar to overview figure in Ras prediction paper)
2. Cross-validation stratified by cancer type (4 folds x 2 replicates)
3. Comparing -omics layers as the predictors/features
4. Description of evaluation metrics/format of volcano-like plots

### Mutation prediction using gene expression data (Figure 2)

1. Results (volcano plot and summary bars) for gene expression only, using Vogelstein cancer gene set
2. Results (volcano plot and summary bars) using 50 most mutated genes in TCGA
3. Results (volcano plot and summary bars) using 50 randomly selected genes

### Gene expression vs. DNA methylation (Figure 3)

1. Compare gene expression, 27K methylation, 450K methylation using Vogelstein genes (volcano plot and summary bars, raw gene expression + 5000 PCs for methylation datasets)
2. Show results for some specific genes (line plots comparing results for various PCA dimensions with raw results)
    - See examples of line plots here: https://docs.google.com/presentation/d/1UQeZTKMsNbZCNDOuECPMQyKVOkJEfq_NOGT5c3aqtL8/edit#slide=id.gbc40d22820_0_5 (slide 5)
    - TP53, ERBB2, IDH1, SETD2 (these cover most of the interesting cases/trends)

### Expression vs. methylation vs. RPPA vs. mutational signatures (Figure 4)

1. Compare all data types using Vogelstein genes (4 volcano plots comparing expression to each other data type)
2. Summary bars comparing all datatypes
3. Maybe show results for some specific genes?

### Multi-omics pilot experiment (Figure 5)

1. Bar plots for each pilot gene (or the interesting subset maybe?)
    - PR for this is forthcoming

### Summary figure? (Figure 6)

1. Could add a heatmap showing performance for each gene in each data type, and highlighting best-performing data type for each gene (+ significance information)? Think about how to do this.
    - This would help to emphasize take-home message (most highlighting would be for gene expression)
