## Results

### Using diverse data modalities to predict cancer alterations

We collected four different readouts from cancer samples in the TCGA Pan-Cancer Atlas, capturing four steps of cellular function that are perturbed by genetic alterations in cancer (Figure {@fig:overview}A).
These included gene expression (RNA-seq data), DNA methylation (27K and 450K Illumina BeadChip arrays), protein abundance (RPPA data), and patterns of somatic mutation (mutational signatures).
To link these diverse data modalities to changes in mutation status, we used elastic net logistic regression to predict the presence or absence of mutations in cancer genes, using these readouts as predictive features (Figure {@fig:overview}B).
We evaluated the resulting mutation status classifiers in a pan-cancer setting, preserving the proportions of each of the 33 cancer types in TCGA for all of 8 train/test splits (4 folds x 2 replicates) in each of 107 cancer genes (Figure {@fig:overview}C).

(talk about framework for comparing classifiers for a given target gene)
<!-- To account for variation between dataset splits in comparing classifiers (i.e. comparing data types for the same target gene), we simply the performance distributions over the 8 train-test splits using a _t_-test -->

![
blah blah figure legend
](images/figure_1.png){#fig:overview}

